src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleAllGoods = async () => {
+    const users1 = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setUsers(users1);
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handle5Goods = async () => {
+    const users2 = await get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setUsers(users2);
+  };
+
+  const handleRedGoods = async () => {
```

Comment:
you can compare your handlers: create one handler that can expect a callback(getAll, get5First, or getRedGoods) that will be called, and the result add to the state